### DISCUSSION TITLE
Clip with pre_capture and post_capture.

### DISCUSSION THREAD

**Discussion by tpjanssen** [blakeblackshear/frigate#3608] [Ask A Question]
Hi,

Is there a way to get a recording through the API that also contains the pre_capture and post_capture? If I download the clip after receiving the MQTT "end" message, the recording does not contain the pre and post captures.

Regards.

**Comment by NickM-27** [comment]
pre_capture and post_capture are just added to the times when the event is saved in the DB. If you use the event.mp4 endpoint then it should include those. Which API are you using to download?

**Comment by tpjanssen** [comment]
I'm using /api/events/<id>/clip.mp4 to download. In the API I don't see anything about event.mp4...

**Reply by NickM-27** [reply]
That's the one I meant, it should be including the pre_capture meaning there's is that much time included from when the object is first visible

**Comment by tpjanssen** [comment]
I've checked again, and something strange is happening. There are three events overlapping:

![image](https://user-images.githubusercontent.com/25168870/183502310-8e3b3e12-5086-40ae-854f-98ae9cfa5f1f.png)

But for some reason I don't get an "end" message for the bottom concatenated clip. Furthermore, sometimes my recordings do not seem to be completely ready. The movie indicates to have the proper length, but it seems to miss 19 seconds or so. How long does it take after the "end" message for the clip to be ready for download?

**Reply by NickM-27** [reply]
> How long does it take after the "end" message for the clip to be ready for download?

I'm not 100% sure but it does take at least 10 seconds since the segment (default is 10 seconds in length) needs to be moved to FS from cache when the segment is finalized.
